### PR TITLE
[Merged by Bors] - feat: port CategoryTheory.Limits.Shapes.Images

### PR DESCRIPTION
**Pull request by mattrobball**
---

- [x] depends on: #2522
[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by semorrison** [comment]
This PR/issue depends on:
* leanprover-community/mathlib4#2522
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

**Comment by semorrison** [comment]
This PR/issue depends on:
* ~~leanprover-community/mathlib4#2522~~
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

File: Mathlib/CategoryTheory/Limits/Shapes/Images.lean [hunk 1]
```diff
@@ -73,58 +73,59 @@ variable {X Y : C} (f : X ⟶ Y)
...
+  /-- The "image" of f -/
+  I : C -- Porting note: violates naming conventions but can't think a better replacement
+  /-- The inclusion of of the image -/
   m : I ⟶ Y
+  /-- `m` is a monomorphism -/
   [m_mono : Mono m]
+  /-- The map to the image -/
   e : X ⟶ I
-  fac : e ≫ m = f := by obviously
+  /-- The factorisation -/
```
Comment (jcommelin) [inline Mathlib/CategoryTheory/Limits/Shapes/Images.lean:84]: There is no need to add docs while porting. So let's not do it for sake of atomic actions :wink: :grinning:

File: Mathlib/CategoryTheory/Limits/Shapes/Images.lean [hunk 2]
```diff
@@ -73,58 +73,59 @@ variable {X Y : C} (f : X ⟶ Y)
...
+  I := X
   m := f
   e := 𝟙 X
 #align category_theory.limits.mono_factorisation.self CategoryTheory.Limits.MonoFactorisation.self
 
 -- I'm not sure we really need this, but the linter says that an inhabited instance
--- ought to exist...
-instance [Mono f] : Inhabited (MonoFactorisation f) :=
-  ⟨self f⟩
+-- ought to exist...asdfafd
```
Comment (jcommelin) [inline Mathlib/CategoryTheory/Limits/Shapes/Images.lean:105]:
```suggestion
-- ought to exist...
```

File: Mathlib/CategoryTheory/Limits/Shapes/Images.lean [hunk 3]
```diff
@@ -178,17 +179,18 @@ variable {f}
 
 /-- Data exhibiting that a given factorisation through a mono is initial. -/
 structure IsImage (F : MonoFactorisation f) where
-  lift : ∀ F' : MonoFactorisation f, F.i ⟶ F'.i
-  lift_fac' : ∀ F' : MonoFactorisation f, lift F' ≫ F'.m = F.m := by obviously
+  /-- The "image" of `F` maps to any other "image" -/
```
Comment (jcommelin) [inline Mathlib/CategoryTheory/Limits/Shapes/Images.lean:182]: Please remove the docs.

File: Mathlib/CategoryTheory/Limits/Shapes/Images.lean [hunk 4]
```diff
@@ -250,9 +251,12 @@ variable (f)
 
 /-- Data exhibiting that a morphism `f` has an image. -/
 structure ImageFactorisation (f : X ⟶ Y) where
-  f : MonoFactorisation f
-  IsImage : IsImage F
+  /-- We can factor `f` into a composition of a monomorphism and another map -/
```
Comment (jcommelin) [inline Mathlib/CategoryTheory/Limits/Shapes/Images.lean:254]: Idem

File: Mathlib/CategoryTheory/Limits/Shapes/Images.lean (hunk 1 above)
Comment (mattrobball) [reply Mathlib/CategoryTheory/Limits/Shapes/Images.lean:84]: `nolint docBlame`?

Comment (mattrobball) [reply Mathlib/CategoryTheory/Limits/Shapes/Images.lean]: Added `@[inherit_doc]` to all the fields

File: Mathlib/CategoryTheory/Limits/Shapes/Images.lean (hunk 2 above)
Comment (mattrobball) [reply Mathlib/CategoryTheory/Limits/Shapes/Images.lean:103]: Fixed

File: Mathlib/CategoryTheory/Limits/Shapes/Images.lean (hunk 3 above)
Comment (mattrobball) [reply Mathlib/CategoryTheory/Limits/Shapes/Images.lean]: Added `@[inherit_doc]` to all the fields

File: Mathlib/CategoryTheory/Limits/Shapes/Images.lean (hunk 4 above)
Comment (mattrobball) [reply Mathlib/CategoryTheory/Limits/Shapes/Images.lean]: Added `@[inherit_doc]` to all the fields

**Pull request closed without merging**